types/scene/wlr_scene.c

```diff
@@ -1010,7 +1014,7 @@ void wlr_scene_shadow_set_corner_radius(struct wlr_scene_shadow *shadow, int cor
 }
 
 void wlr_scene_shadow_set_blur_sigma(struct wlr_scene_shadow *shadow, float blur_sigma) {
-	if (shadow->blur_sigma == blur_sigma) {
+	if (shadow->blur_sigma == blur_sigma || blur_sigma < 0) {
```

Comment:
an aside, but I think we should try to figure out how to handle out of bounds values more formally. I think upstream tends to make an assertion, but we haven't really formed an opinion yet. Curious to hear your thoughts - I think we should assert to match upstream

Reply:
Yeah, lets match upstream! :D